drgn/helpers/common/stack.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Stack
+-----
+
+The ``drgn.helpers.common.stack`` module provides helpers for working with the
+stack trace.
+"""
+
+from drgn import PlatformFlags, StackTrace
+from drgn.helpers.common.memory import identify_address
+
+__all__ = ("print_annotated_stack",)
+
+
+def print_annotated_stack(trace: StackTrace) -> None:
+    """
+    Print the content of the stack trace, identifying symbols and
+    slab objects.
+
+    :param trace: Stack trace to print.
+    """
+    prog = trace.prog
+    # platform must be known if there is a stack trace
+    assert prog.platform is not None
+
+    word_size = 8 if prog.platform.flags & PlatformFlags.IS_64_BIT else 4
+
+    frames = [frame for frame in trace if not frame.interrupted]
```

Comment:
Interrupted frames essentially begin an entirely new stack. So rather than ignoring them, you should treat them almost like separators. So, e.g., if you have frame 0-2 with `interrupted = False`, then frame 3 with `interrupted = True`, then frame 4 with `interrupted = False`, you pretty much have to dump two stacks: one for frames 0-2, and one for frames 3-4.